prow/test/integration/test/webhook_server_test.go

```diff
@@ -17,49 +17,107 @@ limitations under the License.
 package integration
 
 import (
+	"context"
+
+	"io/ioutil"
 	"os/exec"
 	"path/filepath"
 
 	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	"k8s.io/apimachinery/pkg/types"
+	v1 "k8s.io/test-infra/prow/apis/prowjobs/v1"
+	"sigs.k8s.io/yaml"
 )
 
 func cleanUpCluster(prowjob string) error {
-	err := exec.Command("kubectl", "--context=kind-kind-prow-integration", "delete", "prowjob", prowjob).Run()
+	err := exec.Command("kubectl", "--context="+getClusterContext(), "delete", "prowjob", prowjob).Run()
 	if err != nil {
 		return err
 	}
 	return nil
 }
 
 const (
-	validProwJobID   = "test-job2"
+	validProwJobID     = "test-job2"
+	validBareProwJobID = "test-mutate-job"
 )
 
 func TestWebhookServerValidateCluster(t *testing.T) {
+	t.Cleanup(func() {
+		err := cleanUpCluster(validProwJobID)
+		if err != nil {
+			t.Errorf("could not delete prowjob %s", validProwJobID)
+		}
+	})
 	invalidProwJobPath := "./testdata/invalid_prowjob.yaml"
 	validProwJobPath := "./testdata/valid_prowjob.yaml"
 
 	absInvalidProwJobPath, err := filepath.Abs(invalidProwJobPath)
 	if err != nil {
-		t.Errorf("could not find absolute file path")
+		t.Fatalf("could not find absolute file path")
 	}
 	absValidProwJobPath, err := filepath.Abs(validProwJobPath)
 	if err != nil {
-		t.Errorf("could not find absolute file path")
+		t.Fatalf("could not find absolute file path")
 	}
 	//prowjob should fail and produce error
-	err = exec.Command("kubectl", "--context=kind-kind-prow-integration", "apply", "-f", absInvalidProwJobPath).Run()
+	out, err := exec.Command("kubectl", "--context="+getClusterContext(), "apply", "-f", absInvalidProwJobPath).CombinedOutput()
 	if err == nil {
-		t.Errorf("prowjob was not properly validated")
+		t.Fatalf("prowjob was not properly validated. Output: %s", string(out))
 	}
 	//prowjob should be created successfully
-	err = exec.Command("kubectl", "--context=kind-kind-prow-integration", "apply", "-f", absValidProwJobPath).Run()
+	out, err = exec.Command("kubectl", "--context="+getClusterContext(), "apply", "-f", absValidProwJobPath).CombinedOutput()
 	if err != nil {
-		t.Errorf("prowjob was not properly validated")
+		t.Fatalf("prowjob was not properly validated. Error: %v Output: %s", err, string(out))
 	}
-	//cleanup prowjob that was created
-	err = cleanUpCluster(validProwJobID)
+}
+
+func TestWebhookServerMutateProwjob(t *testing.T) {
+	t.Cleanup(func() {
+		err := cleanUpCluster(validBareProwJobID)
+		if err != nil {
+			t.Errorf("could not delete prowjob %s", validProwJobID)
```

Comment:
Cleanup is best effort, not worth fail the test, log it on `Info` level is sufficient